docs/advanced/ObjectSpreadOperator.md

```diff
@@ -0,0 +1,37 @@
+# Object Spread Operator
+
+Since one of the core tenants of Redux is to never mutate state, you'll often find yourself using `Object.assign` to create
+copies of objects with new or updated values. For example, in the `todoApp` below `Object.assign` is used to return a new
+`state` object with an updated `visibilityFilter` property:
+
+```js
+function todoApp(state = initialState, action) {
+  switch (action.type) {
+    case SET_VISIBILITY_FILTER:
+      return Object.assign({}, state, {
+        visibilityFilter: action.filter
+      })
+    default:
+      return state
+  }
+}
+```
+
+While effective, using `Object.assign` can quickly make simple reducers difficult to read given it's rather verbose syntax.
+
+An alternative approach is to use the [object spread syntax](https://github.com/sebmarkbage/ecmascript-rest-spread) proposed for ES7
+which lets you use the spread (`...`) operator to copy enumerable properties from one object to another in a more succinct way. We
+can simplify the `todoApp` example above by using the object spread syntax:
+
+```js
+function todoApp(state = initialState, action) {
+  switch (action.type) {
+    case SET_VISIBILITY_FILTER:
+      return {...state, visibilityFilter: action.filter}
+    default:
+      return state
+  }
+}
+```
+
+Since the object spread syntax is still a Stage 2 proposal for ECMAScript you'll need to use a transpiler such as [babel](http://babeljs.io/) to use it in production. It's available in the [`babel-preset-stage-2`](https://babeljs.io/docs/plugins/preset-stage-2/) plugin as well as a standalone plugin, [`transform-object-read-spread`](http://babeljs.io/docs/plugins/transform-object-rest-spread/).
```

Comment:
Let’s include a more descriptive official recommendation. I would not mention `babel-preset-stage-2` because presets confuse beginners. I would say “You can use your existing `es2015` preset, install `babel-plugin-transform-object-read-spread` package, and add it individually to `plugins` array in your `.babelrc`” and then show how `.babelrc` would look like in this case.